tests/test_lab_config.py

```diff
@@ -22,6 +23,187 @@ def test_ilab_config_show(cli_runner: CliRunner) -> None:
+                "chat_template": None,
+                "host_port": "127.0.0.1:8000",
+                "llama_cpp": {"gpu_layers": -1, "llm_family": "", "max_ctx_size": 4096},
+                "model_path": "/cache/instructlab/models/mistral-7b-instruct-v0.2.Q4_K_M.gguf",
+                "vllm": {
+                    "gpus": None,
+                    "llm_family": "",
+                    "max_startup_attempts": 120,
+                    "vllm_args": [],
+                },
+            },
+        },
+        "serve": {
+            "backend": None,
+            "chat_template": None,
+            "host_port": "127.0.0.1:8000",
+            "llama_cpp": {"gpu_layers": -1, "llm_family": "", "max_ctx_size": 4096},
+            "model_path": "/cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf",
+            "vllm": {
+                "gpus": None,
+                "llm_family": "",
+                "max_startup_attempts": 120,
+                "vllm_args": [],
+            },
+        },
+        "train": {
+            "additional_args": {},
+            "checkpoint_at_epoch": True,
+            "ckpt_output_dir": "/data/instructlab/checkpoints",
+            "data_output_dir": "/data/instructlab/internal",
+            "data_path": "/data/instructlab/datasets",
+            "deepspeed_cpu_offload_optimizer": False,
+            "device": "cpu",
+            "disable_flash_attn": False,
+            "distributed_backend": "deepspeed",
+            "effective_batch_size": 64,
+            "fsdp_cpu_offload_optimizer": False,
+            "is_padding_free": False,
+            "lora_quantize_dtype": "nf4",
+            "lora_rank": 0,
+            "max_batch_len": 5000,
+            "max_seq_len": 4096,
+            "model_path": "instructlab/granite-7b-lab",
+            "nproc_per_node": 1,
+            "num_epochs": 10,
+            "phased_base_dir": "/data/instructlab/phased",
+            "phased_mt_bench_judge": "/cache/instructlab/models/prometheus-eval/prometheus-8x7b-v2.0",
+            "phased_phase1_effective_batch_size": 128,
+            "phased_phase1_num_epochs": 7,
+            "phased_phase1_samples_per_save": 0,
+            "phased_phase2_effective_batch_size": 3840,
+            "phased_phase2_num_epochs": 10,
+            "phased_phase2_samples_per_save": 0,
+            "pipeline": "full",
+            "save_samples": 250000,
+            "training_journal": None,
+        },
+        "version": "1.0.0",
+    }
+
+    result_json = cli_runner.invoke(
+        lab.ilab, ["--config", str(test_config_path), "config", "show", "--json"]
+    )
+
+    assert result_json.exit_code == 0
+    parsed_json = json.loads(result_json.stdout)
+
+    assert parsed_json == expected_json_output
+
+
+def test_ilab_config_show_key_general(
+    cli_runner: CliRunner, testdata_path: pathlib.Path
+) -> None:
+    test_config_path = testdata_path / "default_config.yaml"
+
+    # Expected output as a JSON string for --key general
```

Comment:
I think it would be better to extract this from the config programmatically in this test.